Good Coding Practices/src/js/scripts.js

```diff
@@ -1,44 +1,100 @@
 // piece object
-const piece = (function() {
+const piece = (function () {
+  let defult_pos = null;
   let el = null;
-  const init = function(el) {
+  const init = function (el) {
     this.el = el;
+    this.defult_pos = el.getBoundingClientRect();
   };
-  const moveDelta = function(dx, dy) {
+  const moveDelta = function (dx, dy) {
     const pos = this.el.getBoundingClientRect();
     this.el.style.left = `${pos.left + dx}px`;
     this.el.style.top = `${pos.top + dy}px`;
   };
+
+  const resetPos = function () {
+    this.el.style.left = `${this.defult_pos.left}px`;
+    this.el.style.top = `${this.defult_pos.top}px`;
+  };
+
+  const randPos = function () {
+    let rand_X = Math.floor(Math.random() * window.innerWidth);
+    let rand_Y = Math.floor(Math.random() * window.innerHeight);
+    this.el.style.left = `${rand_X}px`;
+    this.el.style.top = `${rand_Y}px`;
+  };
+
   return {
     init,
-    moveDelta
+    moveDelta,
+    resetPos,
+    randPos
   };
 })();
 
 function handleClick(ev) {
   piece.moveDelta(parseInt(this.dataset.dx), parseInt(this.dataset.dy));
 }
 
+function setBtnClickEvent(btnElement, xDataset, yDataset) {
+  btnElement.dataset.dx = xDataset;
+  btnElement.dataset.dy = yDataset;
+  btnElement.addEventListener("click", handleClick);
+}
+
 function init() {
-  const $btnUp = document.getElementById("btn-up");
-  $btnUp.dataset.dx = 0;
-  $btnUp.dataset.dy = -100;
-  $btnUp.addEventListener("click", handleClick);
-  const $btnRight = document.getElementById("btn-right");
-  $btnRight.dataset.dx = 100;
-  $btnRight.dataset.dy = 0;
-  $btnRight.addEventListener("click", handleClick);
-  const $btnDown = document.getElementById("btn-down");
-  $btnDown.dataset.dx = 0;
-  $btnDown.dataset.dy = 100;
-  $btnDown.addEventListener("click", handleClick);
-  const $btnLeft = document.getElementById("btn-left");
-  $btnLeft.dataset.dx = -100;
-  $btnLeft.dataset.dy = 0;
-  $btnLeft.addEventListener("click", handleClick);
+  const btns = [
+    { id: "btn-up", x: 0, y: -100 },
+    { id: "btn-right", x: 100, y: 0 },
+    { id: "btn-down", x: 0, y: 100 },
+    { id: "btn-left", x: -100, y: 0 }
+  ];
+
+  //set direction btns fuctionalty
+  for (const btn of btns) {
+    const $btnEl = document.getElementById(btn.id);
+    setBtnClickEvent($btnEl, btn.x, btn.y);
+  }
+
+  //set reset btn fuctionalty
+  const $resetBtn = document.getElementById("reset");
+  $resetBtn.addEventListener("click", () => piece.resetPos());
+
+  //set randomize btn fuctionalty
+  const $randomizeBtn = document.getElementById("random");
+  $randomizeBtn.addEventListener("click", () => piece.randPos());
 }
 
 window.addEventListener("DOMContentLoaded", event => {
-  piece.init(document.getElementById("piece"));
+  let piece_el = document.getElementById("piece");
+  piece.init(piece_el);
   init();
+
+  fetch("http://api.apixu.com/v1/current.json?key=dda6e762ae4f41efb7e173552192204&q=tel%20aviv")

```

Comment:
- Can be moved to a separate function (`initTemp`)
- api key should be in a constant